### DISCUSSION TITLE
Version string sorting for variables

### DISCUSSION THREAD

**Discussion by Karsonito** [grafana/grafana#42822] [Dashboards, Panels & Visualizations]
Current numerical sorting of variables in grafana is not useful enough.

For example we get list from data source query:
```
x-2
x-y-2
z-x-2
x-10
```

After numerical (asc) sorting we get:
```
z-x-2
x-2
x-y-2
x-10
```
Which is not suitable because I prefer that other non-numeric chars will be sorted too.

It's better to add version sorting algorithm which implemented in gnu coreutils (sort -V) 
```
x-2
x-10
x-y-2
z-x-2
```

More about version sorting:
https://www.gnu.org/software/coreutils/manual/html_node/Version-sort-is-not-the-same-as-numeric-sort.html
> Version sort (‘sort -V’) first breaks down the string into digits and non-digits parts, and only then compares each part (see annotated example in Version-sort ordering rules).

I didn't found javascript implementation, but found C code
https://github.com/coreutils/gnulib/blob/master/lib/filevercmp.c

Please implement such sorting for variables

**Comment by Karsonito** [comment]
It seams that such topic already started much earlier
https://github.com/grafana/grafana/issues/13179

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
